pkg/kv/kvserver/store_create_replica.go

```diff
@@ -245,42 +245,19 @@ func (s *Store) tryGetOrCreateReplica(
 	//   this newer replica is harmless since it just limits the votes for
```

Comment:
unrelated drive-by question: Is "earlier replica" possible here? Why only "same" or "later"?

Reply:
Earlier might be possible. I think there is a hole here, the tombstone and the RaftReplicaID must really work in unison - if RaftReplicaID is 10 then a tombstone of 5 should be treated as a tombstone of 9.

Let's discuss on Slack, I'd like to get this PR in :-)